src/test/ui/feature-gates/feature-gate-dispatch-from-dyn-cell.rs

```diff
@@ -0,0 +1,9 @@
+// Check that even though Cell: DispatchFromDyn it remains an invalid self parameter type
+
+use std::cell::Cell;
+
+trait Trait{
+    fn cell(self: Cell<&Self>); //~ ERROR invalid `self` parameter type: Cell<&Self>
+}
+
```

Comment:
```suggestion

trait Trait2 {
    fn cell(self: Cell<Self>); //~ ERROR invalid `self` parameter type: Cell<Self>
}
```

Would this make sense to add?

Reply:
If you'd like I'll add.

I don't see what it would be testing though.
I chose `Cell<&Self>` in the test because it could, at least in theory, provide access to `&Self`.
`Cell<Self>` is not able to provide `&Self`.